app/models/volunteer.rb

```diff
@@ -124,66 +129,77 @@ def has_main_region?
   end
 
   def main_region
-    self.regions[0]
+    regions[0]
   end
 
   def region_ids
-    self.regions.collect{ |r| r.id }
+    regions.collect(&:id)
   end
 
-  def admin_region_ids(strict=false)
-    admin_regions(strict).collect { |r| r.id }
+  def admin_region_ids(strict = false)
+    admin_regions(strict).collect(&:id)
   end
 
-  def admin_regions(strict=false)
-    if self.super_admin? and not strict
+  def admin_regions(strict = false)
+    if super_admin? && !strict
       Region.all
     else
-      self.assignments.collect{ |a| a.admin ? a.region : nil }.compact
+      assignments.collect { |a| a.admin ? a.region : nil }.compact
     end
   end
 
-  def in_region? region_id
-    self.region_ids.include? region_id
-  end
-
-  # better first-time experience: if there is only one region, add the user to that one automatically when they sign up
-  def auto_assign_region
-    if Region.count==1 and self.regions.count==0
-      Assignment.add_volunteer_to_region self, Region.first
-      logger.info "Automatically assigned new user to region #{self.regions.first.name}"
-    end
+  def in_region?(region_id)
+    region_ids.include? region_id
   end
 
   def current_absences
-    self.absences.keep_if{ |a| a.start_date < Date.today and a.stop_date > Date.today }
+    today = Date.today
+
+    absences.keep_if do |a|
+      a.start_date < today && a.stop_date > today
+    end
   end
 
   ### CLASS METHODS
 
-  def self.active(region_ids=nil,ndays=90)
-    Volunteer.joins(:logs).select("max(logs.when) as last_log_date,volunteers.*").
-      group("volunteers.id").keep_if{ |v|
-        (Date.parse(v.last_log_date) > Time.zone.today-ndays) and (region_ids.nil? or (v.region_ids & region_ids).length > 0)
-      }
+  def self.active(region_ids = nil, ndays = 90)
+    joins(:logs).
+      select('max(logs.when) as last_log_date,volunteers.*').
+      group('volunteers.id').
+      keep_if do |v|
+        (Date.parse(v.last_log_date) > Time.zone.today - ndays) &&
+          (region_ids.nil? || (v.region_ids & region_ids).length > 0)
+      end
   end
 
-  def self.inactive(region_ids=nil)
-    Volunteer.find_all_by_active(false).keep_if{ |v| (region_ids.nil? or (v.region_ids & region_ids).length > 0) }
+  def self.inactive(region_ids = nil)
+    find_all_by_active(false).
+      keep_if do |v|
+        (region_ids.nil? || (v.region_ids & region_ids).length > 0)
+      end
   end
 
-  def self.all_for_region region_id
-    self.includes(:regions).where(:regions=>{:id=>region_id}).compact
+  def self.all_for_region(region_id)
+    includes(:regions).
+      where(regions: { id: region_id }).
+      compact
   end
 
-
   private
 
+  # better first-time experience: if there is only one region, add the user to
+  # that one automatically when they sign up
+  def auto_assign_region
+    return unless Region.count == 1 && regions.count.zero?
```

Comment:
Oh interesting. I think I lean toward the `.count == 1` solution just because it uses a SQL `COUNT()` as opposed to loading all of the records into memory and looping over them to see if exactly one is truthy. What do you think?

```ruby
[10] pry(main)> Region.all.one?
  Region Load (0.7ms)  SELECT "regions".* FROM "regions"
=> true
[11] pry(main)> Region.count == 1
   (0.5ms)  SELECT COUNT(*) FROM "regions"
=> true
```